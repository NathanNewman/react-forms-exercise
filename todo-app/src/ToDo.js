import React from "react";

const ToDo = ({ id, name, handleRemove }) => {
  const remove = () => handleRemove(id);
  return (
    <div>
      <p>
        {name} <button onClick={remove}>X</button>
      </p>
    </div>
  );
};

export default ToDo;
