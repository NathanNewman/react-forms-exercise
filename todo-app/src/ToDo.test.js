import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import { toBeInTheDocument } from "@testing-library/jest-dom";
import ToDo from "./ToDo";

afterEach(cleanup);

// smoke test
it("renders without crashing", function() {
  render(<ToDo />);
});

// snapshot
test("it matches snapshot", () => {
  const { asFragment } = render(<ToDo />);
  expect(asFragment()).toMatchSnapshot();
});

it("render a todo", () => {
  const { getByText } = render(<ToDo name="test" />);
  const todo = getByText("test");
  expect(todo).toBeInTheDocument();
});
