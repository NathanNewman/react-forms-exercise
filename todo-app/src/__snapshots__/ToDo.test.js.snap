// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
       
      <button>
        X
      </button>
    </p>
  </div>
</DocumentFragment>
`;
